gradebook/src/GradeBook/book.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+
+/*
+The class Book packages the blueprint book provides ways to:
+1 store the grades. 
+2 add to the given gradebook obj
+3 welcomes the user from args
+4 Assign name to the gradebook
+ */
+
+namespace GradeBook
+{
+    public class book
+    {
+        List<double> grades;
+        string name;
+        double highGrade = double.MinValue;
+        double lowGrade = double.MaxValue;
+        double avg = 0.0;
+
+        //Constructor
+        public book(string name, string[] args) 
+        {
+            grades = new List<double>() {1,22,56,33.2,45,30,32};
+            this.name = name;
+            welcomeUser(args);
+        }
+
+        public void calculateStats()
+        {
+            foreach(double gd in this.grades)
+            {
+                avg += gd;
+                highGrade = Math.Max(highGrade,gd);
+                lowGrade = Math.Min(lowGrade, gd);
+            }
+            avg /= this.grades.Count;
+        }
+
+        // show the current statistics
+        public void showStatistics() 
+        {
+            calculateStats();
+            Console.WriteLine($"Min grade - {lowGrade}\nMax Grade- {highGrade}\nAvg Grade - {avg}");
+        }
+
+        // Welcomes the user
+        private void welcomeUser(string[] args)
+        {
+            string user = "User";
+            if(args.Length>0)
```

Comment:
I would say this should just take a `user` parameter.  You may also want to check `string.IsNullOrWhitespace` and throw an `ArgumentException` if the parameter is invalid.

Reply:
Yeah. I can do that. However, I wanted to have a default just so that if the user misses to provide their name, we still have a valid string in-place. But I can add error handling for Null or white space.

Reply:
I would suggest handling verification and defaulting before you get to this point.  In general I think you're best off validating/defaulting parameters as soon as they're given to you - in the constructor in this case.  That makes it easier for someone using the class to tell something is wrong as early as possible.